manager/orchestrator/update/updater.go

```diff
@@ -320,7 +321,17 @@ func (u *Updater) worker(ctx context.Context, queue <-chan orchestrator.Slot, up
 			cleanTask   *api.Task
 		)
 		for _, t := range slot {
-			if !u.isTaskDirty(t) {
+			// Don't need a new task if:
+			// 1. Either the task spec didn't change at all, or
+			// 2. The placement constraints alone changed and the node currently assigned
+			// can satisfy the changed constraints.
+			if u.isTaskPlacementConstraintsOnlyDirty(t) || !u.isTaskDirty(t) {
+				if u.isTaskPlacementConstraintsOnlyDirty(t) && u.nodeMatches(t) {
```

Comment:
I think a new task is needed if the node does not match the changed constraints

Reply:
ahh yes, I'll just move this condition up and add a unit-test.